modules/system/assets/ui/js/popup.js

```diff
@@ -207,8 +209,22 @@
         if (this.options.adaptiveHeight)
             modalDialog.addClass('adaptive-height')
 
+        if (this.options.cssClass)
+            modalDialog.addClass(this.options.cssClass)
+
         if (this.options.zIndex !== null)
             modal.css('z-index', this.options.zIndex + 20)
+        
+        if (this.options.allowDismiss) {
+            modal.on('mousedown', function(e) {
+                const target = e.target;
+                if (target.classList.contains('control-popup')) {
+                    modal.hide()
+                    $('.popup-backdrop').remove()
+                    $(document.body).removeClass('modal-open')
+                }
+            });
+        }
```

Comment:
Were you able to get the change monitoring to work?

Reply:
No there is no change monitor.
I remember trying to add it but it was looking at the main record instead of the related record

Reply:
@jaxwilko is this something you could take a look at? I'm concerned that if we add this functionality without that as a failsafe it could make accidental data loss for users too easy when in theory it shouldn't be too difficult to add the change monitoring before we merge this.

Reply:
@AIC-BV were you looking at https://wintercms.com/docs/v1.2/ui/script/input-monitor? (https://github.com/wintercms/winter/blob/develop/modules/system/assets/ui/js/input.monitor.js)

Reply:
Yes